Credential Offer enhancements + Response mode cleanup (#916)
italia/eidas-it-wallet-docs: docs/en/test-plans-wallet-provider.rst

     - Wallet Instance builds and verifies the full Trust Chain from the Credential Issuer through Intermediaries to the root Trust Anchor, ensuring each signature is valid and confirms the root.
   * - WP_047
     - Issuance, Interoperability
     - Third-Party-Initiated flow: Credential Offer QR scan & decode

>      - In an Third-Party-Initiated cross-device flow, Wallet Instance successfully scans the QR code, and decodes the ``credential_offer`` URL.

Reviewer: ```suggestion
     - In a Third-Party-Initiated cross-device flow, Wallet Instance successfully scans the QR code, and retrieves the ``credential_offer`` or the ``credential_offer_uri`` parameter.
```

   * - WP_048
     - Issuance, Interoperability
     - Third-Party-Initiated flow: Credential Offer parsing

>      - In an Third-Party-Initiated flow, Wallet Instance extracts and validates all required parameters present (``iss``, ``credential_offer``, etc.) in the URL-decoded ``credential_offer``.

Reviewer: ```suggestion
     - In a Third-Party-Initiated flow, Wallet Instance extracts and validates all required parameters present in the Credential Offer.
```
